brain/compartmentReportView.cpp

```diff
@@ -99,6 +99,36 @@ std::future<brion::Frames> CompartmentReportView::load(double start, double end)
     return _impl->report->loadFrames(start, end);
 }
 
+std::future<brion::Frames> CompartmentReportView::load(double start, double end,
+                                                       double stride)
+{
+    if (end <= start)
+        throw std::logic_error("Invalid interval");
+
+    if (stride < _impl->report->getTimestep())
+        throw std::logic_error("Invalid stride");
+
+    start = std::max(start, _impl->report->getStartTime());
+    end = std::min(end, _impl->report->getEndTime());
+
+    auto task = [this, start, end, stride] {
+
+        brion::Frames frames;
+        frames.timeStamps.reset(new brion::doubles);
+        frames.data.reset(new floats);
+        for (double t = start; t < end; t += stride)
```

Comment:
This loop is troublesome expressed this way, there are going to be cases of rounding errors.
In fact using a floating point stride is troublesome in itself.
It's better to write a while loop that increases a counter, the condition will be start + i * stride < end.
With stride being a integer with the number of frames to advance, if would be start + timestep*0.5 + i * stride * timestep < end